### PR TITLE
feat: (Day) Add option to allow safe zones to consider any player friendly

### LINKED ISSUE #2070: enhancement: Allow option for friendly-fire in safe zones for non party/guildies
### Description

Some games may want to have a heavier emphasis on teamwork and enable friendly casting on safe maps for players that aren't in their party/guild. This would enable that.

### Intended Use-Case

Allow more configuration for developers.

### Duplicate Check

- [X] This enhancement idea is not a duplicate to the best of my knowledge.

### PR DESCRIPTION
**Pull request by AlexVild**
Resolves #2070

### CODE REVIEW

File: Intersect (Core)/Config/CombatOptions.cs [hunk 1]
```diff
@@ -68,5 +68,10 @@ public partial class CombatOptions
         /// If enabled, this allows entities to turn around while casting
         /// </summary>
         public bool EnableTurnAroundWhileCasting = false;
+
+        /// <summary>
+        /// If enabled, this allows players to cast friendly spells on players who aren't in their guild or party
+        /// </summary>
+        public bool EnableSafeZoneAllPlayersFriendly = false;
```
Comment (lodicolo) [inline Intersect (Core)/Config/CombatOptions.cs:75]: Property please

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect (Core)/Config/CombatOptions.cs (hunk 1 above)
Comment (AlexVild) [reply Intersect (Core)/Config/CombatOptions.cs]: Done

File: Intersect (Core)/Config/CombatOptions.cs [hunk 2]
```diff
@@ -68,5 +68,10 @@ public partial class CombatOptions
         /// If enabled, this allows entities to turn around while casting
         /// </summary>
         public bool EnableTurnAroundWhileCasting = false;
+
+        /// <summary>
+        /// If enabled, this allows players to cast friendly spells on players who aren't in their guild or party
+        /// </summary>
+        public bool EnableSafeZoneAllPlayersFriendly { get; set; } = false;
```
Comment (lodicolo) [inline Intersect (Core)/Config/CombatOptions.cs:75]: `EnableAllPlayersFriendlyInSafeZone`

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect (Core)/Config/CombatOptions.cs (hunk 2 above)
Comment (AlexVild) [reply Intersect (Core)/Config/CombatOptions.cs]: done

**Review by lodicolo — APPROVED** [review]

**Pull request merged**